app/validators/base_validator.rb

```diff
@@ -234,4 +238,8 @@ def looks_like_a_case_number?(attribute)
     return if attr_blank?(attribute)
     @record.__send__(attribute).match?(CASE_NUMBER_OR_URN_PATTERN)
   end
+
+  def allow_future_dates
+    ENV.fetch('ALLOW_FUTURE_DATES', nil).eql?('true') && !ENV.fetch('ENV', nil).eql?('production')
```

Comment:
A couple of suggestions that are matters of opinion, so I don't have objections if you decide against them;

```suggestion
    return false if ENV.fetch('ENV', nil).eql?('production')

    ActiveModel::Type::Boolean.new.cast(ENV.fetch('ALLOW_FUTURE_DATES', false))
```

1. Using a guard clause to exclude the 'production' case up-front.
2. Rails has a helper to parse boolean values. It is really intended for http parameters but I believe it can be used here too.

Reply:
Both good points. I've added the guard. I found that CCCD defines it's own [helper](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/blob/master/lib/extensions/string_extension.rb) for parsing boolean strings, so I've used that instead as it's a bit cleaner.

Reply:
Hmmm, I'd prefer to get away from those monkey-patched custom helpers, although they are fairly deeply embedded. I think this one was put in before the standard Rails way of doing it existed.